versionparse/lib.go

```diff
@@ -20,47 +20,132 @@ import (
 	"io/ioutil"
 	"os"
 	"regexp"
+	"strconv"
+	"strings"
 
-	version "github.com/hashicorp/go-version"
 	log "github.com/sirupsen/logrus"
 )
 
 var (
-	kernelVersionRegexp = regexp.MustCompile(`Linux version (\d+\.\d+\.\d+)`)
+	kernelVersionRegexp = regexp.MustCompile(`Linux version (\d+\.\d+\.\d+(?:-\d+)?)`)
 )
 
-func MustParseVersion(v string) *version.Version {
-	ver, err := version.NewVersion(v)
+type Version struct {
+	versionSlice []int
+	versionStr   string
+}
+
+func (v *Version) String() string {
+	return v.versionStr
+}
+
+func NewVersion(ver string) (*Version, error) {
+	var v Version
+	var err error
+	v.versionSlice, err = convertVersionToIntSlice(ver)
+	v.versionStr = ver
+	return &v, err
+}
+
+func MustParseVersion(v string) *Version {
+	ver, err := NewVersion(v)
 	if err != nil {
 		log.WithError(err).Panic("Failed to parse version.")
 	}
 	return ver
 }
 
+func (v *Version) Compare(other *Version) int {
+	vlen := len(v.versionSlice)
+	olen := len(other.versionSlice)
+	compLen := vlen
+	if compLen > olen {
+		compLen = olen
+	}
+	for index := 0; index < compLen; index++ {
+		if v.versionSlice[index] == other.versionSlice[index] {
+			continue
+		}
+		if v.versionSlice[index] > other.versionSlice[index] {
+			return 1
+		}
+		if v.versionSlice[index] < other.versionSlice[index] {
+			return -1
+		}
+	}
+	if vlen < olen {
+		return -1
+	}
+	if vlen > olen {
+		return 1
+	}
+	return 0
+}
+
+func convertVersionToIntSlice(s string) ([]int, error) {
+	var splitRe = regexp.MustCompile(`[\.-]`)
```

Comment:
Suggest moving the regex creation out of the function so it only gets compiled once.